### ISSUE TITLE
Register TypeSpec on github/linguist

### ISSUE THREAD

**Issue by timotheeguerin** [microsoft/typespec#1040]
Issue tracking the progress towards being able to register tsp as a known github language. 

Requirements: 
- 200 unique repo are using cadl on github
https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language

Gitub search result showing usage https://cs.github.com/?scopeName=All+repos&scope=&q=path%3A*.tsp

PR adding support https://github.com/github-linguist/linguist/pull/6775, currently waiting on more usage of TypeSpec

**Comment by connorjs** [comment]
If we want a tracking request on the linguist side, we could create one with their issue template: https://github.com/github-linguist/linguist/issues/new/choose

I’ll defer to y’all on that though. We can also update this issue’s title to TypeSpec.

Here’s a more accurate (I think) GitHub search for `tsp` files: https://github.com/search?q=path%3A*.tsp+%2F%5Enamespace%2F&type=code - checking for `^namespace` filters out a lot of non TypeSpec usages.

**Comment by timotheeguerin** [comment]
We were waiting on getting the 200 unique repo uses, good idea on the more targeted search though we are now getting 7k repos which seems wrong

**Comment by bterlson** [comment]
Adding `import` and `namespace` to the list results in a more accurate count, though still have to wade through forks and such of our repos.

**Comment by connorjs** [comment]
Yeah, I don’t see a way to say "run a code search, then uniq the repos" (but I bet that’s easy enough to script).

Searching for `main.tsp` may provide the least results: <https://github.com/search?q=path%3Amain.tsp&type=code> (shows 239 files at time of writing)

We could filter out Microsoft organizations if desired: <https://github.com/search?q=path%3Amain.tsp+NOT+org%3AAzure+NOT+org%3Amicrosoft+&type=code> (shows 210 files at time of writing)

I can cook up a more accurate script later this week. I’ve always wanted to check out a GitHub API.

**Comment by timotheeguerin** [comment]
Current query that I think is quite accurate, filtering out forks and searching for the import keyword which is more likely to exists than `namespace`

https://github.com/search?q=NOT+is:fork+path:*.tsp+/%5Eimport/&type=code

**Comment by timotheeguerin** [comment]
Doing a rough count of the result I get 20-30 different repos today so still quite short of the 200 required. Hopefully with the announcement and recent interest we would get more

**Comment by timotheeguerin** [comment]
PR requesting https://github.com/github-linguist/linguist/pull/6775

**Comment by timotheeguerin** [comment]
this is done, linguist pr was merged and TypeSpec syntax highlighting is on github
